llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -667,14 +685,25 @@ bool ScalarizerVisitor::splitBinary(Instruction &I, const Splitter &Split) {
 bool ScalarizerVisitor::isTriviallyScalarizable(Intrinsic::ID ID) {
   if (isTriviallyVectorizable(ID))
     return true;
+  // TODO: investigate vectorizable frexp
+  switch (ID) {
+  case Intrinsic::frexp:
```

Comment:
Making this vectorizable was out of scope for this work so did not add `frexp` to `isTriviallyVectorizable`. We should do a follow up task for this.

Reply:
I'll add https://github.com/llvm/llvm-project/issues/112408 to the TODO on line 688.